File: fix_imports.py
import os
import re

replacements = [
  (r'@/components/ThemeProvider', '@/shared/ui/ThemeProvider'),
  (r'@/lib/utils', '@/shared/lib/utils'),
  (r'@/lib/data', '@/entities/testimonial/model/data'),
  (r'@/lib/projects-data', '@/entities/project/model/projects-data'),
  (r'@/lib/github-data', '@/entities/project/model/github-data'),
  (r'@/lib/roadmap-data', '@/entities/skill/model/roadmap-data'),
  (r'@/lib/tech-icons', '@/shared/lib/tech-icons'),
  (r'@/lib/db', '@/shared/lib/db'),
  (r'@/components/SkillTree', '@/entities/skill/ui/SkillTree'),
  (r'@/components/ProjectArchitectureDiagram', '@/entities/project/ui/ProjectArchitectureDiagram'),
  (r'@/components/ProjectGalleryShowcase', '@/entities/project/ui/ProjectGalleryShowcase'),
  (r'@/components/ProjectLifecycleTracker', '@/entities/project/ui/ProjectLifecycleTracker'),
  (r'@/components/BookItem', '@/entities/project/ui/BookItem'),
  (r'@/hooks/use-mobile', '@/shared/lib/hooks/use-mobile'),
]

for root, _, files in os.walk('./src'):
    for file in files:
        if file.endswith('.ts') or file.endswith('.tsx'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            orig = content
            for old, new in replacements:
                content = re.sub(old, new, content)
                
            if content != orig:
                with open(path, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f"Updated {path}")

File: fix_imports.js
const fs = require('fs');
const path = require('path');

const replacements = [
  { from: /@\/components\/ThemeProvider/g, to: '@/shared/ui/ThemeProvider' },
  { from: /@\/lib\/utils/g, to: '@/shared/lib/utils' },
  { from: /@\/lib\/data/g, to: '@/entities/testimonial/model/data' },
  { from: /@\/lib\/projects-data/g, to: '@/entities/project/model/projects-data' },
  { from: /@\/lib\/github-data/g, to: '@/entities/project/model/github-data' },
  { from: /@\/lib\/roadmap-data/g, to: '@/entities/skill/model/roadmap-data' },
  { from: /@\/lib\/tech-icons/g, to: '@/shared/lib/tech-icons' },
  { from: /@\/lib\/db/g, to: '@/shared/lib/db' },
  { from: /@\/components\/SkillTree/g, to: '@/entities/skill/ui/SkillTree' },
  { from: /@\/components\/ProjectArchitectureDiagram/g, to: '@/entities/project/ui/ProjectArchitectureDiagram' },
  { from: /@\/components\/ProjectGalleryShowcase/g, to: '@/entities/project/ui/ProjectGalleryShowcase' },
  { from: /@\/components\/ProjectLifecycleTracker/g, to: '@/entities/project/ui/ProjectLifecycleTracker' },
  { from: /@\/components\/BookItem/g, to: '@/entities/project/ui/BookItem' },
  { from: /@\/hooks\/use-mobile/g, to: '@/shared/lib/hooks/use-mobile' },
];

function walk(dir) {
  let results = [];
  const list = fs.readdirSync(dir);
  list.forEach(function(file) {
    file = dir + '/' + file;
    const stat = fs.statSync(file);
    if (stat && stat.isDirectory()) {
      results = results.concat(walk(file));
    } else {
      if (file.endsWith('.ts') || file.endsWith('.tsx')) {
        results.push(file);
      }
    }
  });
  return results;
}

const files = walk('./src');
files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let original = content;
  replacements.forEach(rep => {
    content = content.replace(rep.from, rep.to);
  });
  if (content !== original) {
    fs.writeFileSync(file, content, 'utf8');
    console.log('Updated ' + file);
  }
});
